### ISSUE TITLE
runserver and daphne gives error after upgrade

### ISSUE THREAD

**Issue by rajasimon** [andrewgodwin/channels#73]
I upgraded the channels version today. And look like multiple things not working correctly. After reading the docs and I found need to update `asgi.py`  file and running daphne gives me below error..

```
(venv)simon@simon-OptiPlex-780:~/Workspace/buildsys$ daphne -b 127.0.0.1 -p 9000 buildsys.asgi:channel_layer 
Traceback (most recent call last):
  File "/home/simon/Workspace/venv/bin/daphne", line 9, in <module>
    load_entry_point('daphne==0.9', 'console_scripts', 'daphne')()
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/daphne/cli.py", line 53, in entrypoint
    cls().run(sys.argv[1:])
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/daphne/cli.py", line 74, in run
    channel_layer = importlib.import_module(module_path)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "./buildsys/asgi.py", line 6, in <module>
    channel_layer = get_channel_layer()
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/channels/asgi.py", line 83, in get_channel_layer
    django.setup(set_prefix=False)
TypeError: setup() takes no arguments (1 given)
```

And running runserver also gives me below error :>

```
Unhandled Error
Traceback (most recent call last):
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/channels/management/commands/runserver.py", line 81, in inner_run
    action_logger=self.log_action,
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/daphne/server.py", line 23, in run
    reactor.run(installSignalHandlers=self.signal_handlers)
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/twisted/internet/base.py", line 825, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/daphne/server.py", line 40, in backend_reader
    self.factory.dispatch_reply(channel, message)
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/daphne/http_protocol.py", line 197, in dispatch_reply
    self.reply_protocols[channel].serverResponse(message)
  File "/home/simon/Workspace/venv/local/lib/python2.7/site-packages/daphne/http_protocol.py", line 136, in serverResponse
    raise ValueError("Got multiple Response messages!")
exceptions.ValueError: Got multiple Response messages!
```

**Comment by andrewgodwin** [comment]
What version of Django, channels, Daphne, asgiref are you using, and what's your channel layer setting look like?

**Comment by rajasimon** [comment]
Pip freeze

```
Django==1.9.2
Twisted==15.5.0
argparse==1.2.1
asgiref==0.9
autobahn==0.12.1
channels==0.9.1
daphne==0.9
psycopg2==2.6.1
pytz==2015.7
six==1.10.0
txaio==2.2.1
wsgiref==0.1.2
zope.interface==4.1.3
```

And my channel layer settings look like this ...

```
# In settings.py
CHANNEL_LAYERS = {
    "default": {
        "BACKEND": "channels.database_layer.DatabaseChannelLayer",
        "ROUTING": "buildsys.routing.channel_routing",
    },
}
```

**Comment by rajasimon** [comment]
I don't know why but looks like " Got multiple Response messages " not coming right now when running runserver ...

And changing    asgi.py line_numer : 83 `django.setup(set_prefix=False)` this to `django.setup()` will work..and I got connected message in my chrome console.  But in my terminal I got this

```
IntegrityError: null value in column "expiry" violates not-null constraint
DETAIL:  Failing row contains (1243, chat, !websocket.send.kfVqXOSu, null).
```

**Comment by andrewgodwin** [comment]
You shouldn't need django.setup in ASGI.py - where in the docs did I put that? I should fix it.

And for the database, delete your channels tables; the schema changed a bit since the earlier version, and I didn't do migrations since the user base was very low then.

**Comment by rajasimon** [comment]
No actually I was saying that In the fresh installation without editing that 83 line it throws error by default.

**Comment by andrewgodwin** [comment]
Oh, yes! I've fixed that in fe352542cc6a53639eb4d57dcf445f238f2f6c2e.

The "multiple response messages" problem turned out to be an issue with the database backend duplicating messages, the new version now uses transactions to stop that - so I'm closing this, since I think these are both fixed. Reopen if it's not.
